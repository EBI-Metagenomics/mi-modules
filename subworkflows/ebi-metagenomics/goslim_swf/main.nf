
include { GENERATEGAF           } from '../../../modules/ebi-metagenomics/generategaf/main'
include { OWLTOOLS              } from '../../../modules/ebi-metagenomics/owltools/main'
include { SUMMARISEGOSLIMS      } from '../../../modules/ebi-metagenomics/summarisegoslims/main'

workflow GOSLIM_SWF {

    take:
    ch_ips     // channel: [ val(meta), path(tsv) ]
    go_obo     // channel: path(obo)
    goslim_ids // channel: path(txt)
    go_banding // channel: path(txt)

    main:

    ch_versions = Channel.empty()


    GENERATEGAF( ch_ips )
    ch_versions = ch_versions.mix(GENERATEGAF.out.versions.first())

    OWLTOOLS ( 
                GENERATEGAF.out.gaf,
                go_obo,
                goslim_ids
    )
    ch_versions = ch_versions.mix(OWLTOOLS.out.versions.first())

    SUMMARISEGOSLIMS (
        ch_ips,
        OWLTOOLS.out.gaf,
        go_obo,
        go_banding
    )
    ch_versions = ch_versions.mix(SUMMARISEGOSLIMS.out.versions.first())

    emit:
    go_summary            = SUMMARISEGOSLIMS.out.go_summary     // channel: [ val(meta), path(csv) ]
    goslim_summary        = SUMMARISEGOSLIMS.out.goslim_summary // channel: [ val(meta), path(csv) ]
    versions              = ch_versions                         // channel: [ versions.yml ]
}

