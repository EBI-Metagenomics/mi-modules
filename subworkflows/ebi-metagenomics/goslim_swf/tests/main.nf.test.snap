{
    "goslim_swf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary.csv:md5,ddf9aa47da0a848795fffb56d9ac3c96"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary_slim.csv:md5,ce50e0507a12ef0d70bd41b14926c5d5"
                    ]
                ],
                "2": [
                    "versions.yml:md5,9803d78712ef1091ff5f785e34209e66",
                    "versions.yml:md5,db47fe42ea02f1cd3ecc4074727e8972",
                    "versions.yml:md5,f7674bdfeee9a4d98dbb33e474dd7a68"
                ],
                "go_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary.csv:md5,ddf9aa47da0a848795fffb56d9ac3c96"
                    ]
                ],
                "goslim_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary_slim.csv:md5,ce50e0507a12ef0d70bd41b14926c5d5"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9803d78712ef1091ff5f785e34209e66",
                    "versions.yml:md5,db47fe42ea02f1cd3ecc4074727e8972",
                    "versions.yml:md5,f7674bdfeee9a4d98dbb33e474dd7a68"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-28T22:32:17.095895573"
    }
}